test/Microsoft.TestPlatform.TestUtilities/IntegrationTestBase.cs

```diff
@@ -643,15 +665,23 @@ private void ExecuteApplication(string path, string args, out string stdOut, out
 
                 stopwatch.Stop();
 
-                Console.WriteLine($"IntegrationTestBase.Execute: Total execution time: {stopwatch.Elapsed.Duration()}");
+                WriteLine($"IntegrationTestBase.Execute: Total execution time: {stopwatch.Elapsed.Duration()}");
 
                 stdError = stderrBuffer.ToString();
                 stdOut = stdoutBuffer.ToString();
                 exitCode = process.ExitCode;
 
-                Console.WriteLine("IntegrationTestBase.Execute: stdError = {0}", stdError);
-                Console.WriteLine("IntegrationTestBase.Execute: stdOut = {0}", stdOut);
-                Console.WriteLine($"IntegrationTestBase.Execute: Stopped {executableName}. Exit code = {0}", exitCode);
+                WriteLine("IntegrationTestBase.Execute: stdError = {0}", stdError);
+                WriteLine("IntegrationTestBase.Execute: stdOut = {0}", stdOut);
+                WriteLine($"IntegrationTestBase.Execute: Stopped {executableName}. Exit code = {0}", exitCode);
+            }
+        }
+
+        private void WriteLine(string format, params object[] arg)
+        {
+            if (TestContext == null)
```

Comment:
If test context is not null, are we not going to report the output to console anymore?

Reply:
If we have a context, I redirect all output to a file and attach it to the test result itself. (or I was hoping to :))